utils/index.d.ts

```diff
@@ -25,6 +25,14 @@ export declare class AzExtTreeDataProvider implements TreeDataProvider<AzExtTree
      */
     public constructor(rootTreeItem: AzExtParentTreeItem, loadMoreCommandId: string);
 
+    /**
+     * Allows the data provider to listen to the {@link TreeView.onDidExpandElement} and {@link TreeView.onDidCollapseElement} events.
+     *
+     * This will ensure that the {@link AzExtParentTreeItem.collapsibleState} property is updated for the element.
+     * @param treeView The {@link TreeView} associated with this data provider
+     */
+    public setupCollapsibleStateListeners(treeView: TreeView<AzExtTreeItem>): Disposable;
```

Comment:
Seems like this should be implemented in an internal fashion; i.e. creating the tree data provider would also create these listeners and it would automatically keep the state of the object up-to-date, without consumers needing to call this. That leaves dealing with the disposables which I'm not immediately sure how to do.